### ISSUE TITLE
Color picker auto close in modal

### ISSUE THREAD

**Issue by valpuia** [laravel-filament/filament#8661]
### Package

filament/filament

### Package Version

v3.0.59

### Laravel Version

10.24.0

### Livewire Version

v3.0.5

### PHP Version

8.2

### Problem description

Following https://github.com/filamentphp/filament/issues/8614#issuecomment-1728675742

The color picker in modal (--simple) resource is hard to pick, it auto close the color panel(?) after click.
[Screencast from 09-22-2023 09:15:19 AM.webm](https://github.com/filamentphp/filament/assets/28250303/e4756a49-9ac3-4201-bd28-019446bdd243)

*Note: The latest version is point to `3.0.57` but when i update I got `3.0.59`*

### Expected behavior

Easy way to pick color in simple (like normal resource)

### Steps to reproduce

Clone, install, login using admin@example.com and password credentials
Go to color resource, open modal and play around

### Reproduction repository

https://github.com/valpuia/color-bug

**Comment by ilpav12** [comment]
Trying to fix this I noticed that removing the `focusing` event listener attached to the document fixes this issue:
![image](https://github.com/filamentphp/filament/assets/35311720/761195fc-23f5-4f4f-8576-adfd971ae960)
The event listener points to this code in `livewire.js`:
```js
var checkFocusIn = function checkFocusIn2(e) {
  var target = getActualTarget(e);
  var targetContained = findContainerIndex(target) >= 0;
  if (targetContained || target instanceof Document) {
    if (targetContained) {
      state.mostRecentlyFocusedNode = target;
    }
  } else {
    e.stopImmediatePropagation();
    tryFocus(state.mostRecentlyFocusedNode || getInitialFocusNode());
  }
};
```
Every time the modal is opened this event listener is added and removed when the modal is closed.
Hopefully this helps creating the PR

**Comment by danharrin** [comment]
Downloaded your repository and updated it to the latest Filament, and got the issue. Then I updated Filament, and I no longer encounter this issue

https://github.com/filamentphp/filament/assets/41773797/a95739c8-70a9-4223-ba3e-bb17fc12c8ea

**Comment by taha-abujrad** [comment]
@danharrin
I have downloaded the above repo and got the auto close problem of the modal, then i updated filament
from: ^3.0-stable
to: ^v3.0.91
and the problem is still occuring.
how you managed to solve it in the above video.
is there a specific version?

**Comment by danharrin** [comment]
Make sure you run `filament:upgrade` and your browser cache is clear

**Comment by taha-abujrad** [comment]
Thank you for replying @danharrin
I have upgraded filament to the latest version
and run all the cache cleaning commands
but the problem is still happening.

**Comment by danharrin** [comment]
Without being able to replicate the issue, I can't fix it

**Comment by taha-abujrad** [comment]
@danharrin
I have used the above repo from [valpuia](https://github.com/valpuia/color-bug) as a reference and do as you have said, howver i could not get it fixed.
Could you please push your changes on the same repo so that i can see them.

Also i want you to include the detailed commands you have ran in your terminal in your comment.

**Comment by danharrin** [comment]
Please do the opposite, open a new issue, with a new Laravel project, install the latest Filament and reproduce the bug

**Comment by ilpav12** [comment]
@danharrin I can confirm that the bug is still present on the latest v3.0.92 using this repository and Windows with Chrome, Edge and Firefox.

**Comment by taha-abujrad** [comment]
@danharrin
here is a repo that you can use to replicate the issue
https://github.com/taha-abujrad/filament-color-bug
please comment detailed commands you have ran in your terminal, and the browser (if any)
if you manged to fix the issue, please do a pull request on the same repo or @ilpav12's repo if he is ok with that.

**Comment by taha-abujrad** [comment]
@danharrin any updates

**Comment by duongductrong** [comment]
https://github.com/filamentphp/filament/assets/39333905/8bde43fc-547c-45ea-97ff-9ed641896c9e

its still not working, any updates for this bug, guys?

```
"filament/filament": "^3.0-stable",
"filament/infolists": "^3.0-stable",
"filament/notifications": "^3.0-stable",
```

**Comment by ivanBell199** [comment]
I have the same issue on Windows with Filament verson 3.0.95

**Comment by saschagysel** [comment]
Same issue with newest Filament 3.0.99. Its happening for color pickers in modals for me.

**Comment by SimonBarrettACT** [comment]
I am using v3.0.102 and have the same issue with Slide Overs.

When the colour-picker popup is clicked to select a colour, it closes.

**Comment by danharrin** [comment]
Please can someone pinpoint the exact version where this breaks, because it worked before

**Comment by saschagysel** [comment]
I wasn't successful yet in finding the exact version (didn't have enough time yet). If someone could support in finding the breaking version that would be great, so Dan can fix this.

It must be a release after 3.0.82 as Dan I tested it then and it worked.

**Comment by ivanBell199** [comment]
On my computer it doesn't work on 3.0.82 and any other version. I tried 3.0.81, 3.0.82, 3.0.83, a few latest versions and even 3.0.50 and 3.0.60

**Comment by danharrin** [comment]
Should be fixed again by #10142
